src/service/viewer-impl.js

```diff
@@ -860,12 +860,33 @@ export class Viewer {
     if (!this.hasCapability('fragment')) {
       return Promise.resolve('');
     }
-    return this.sendMessageUnreliable_('fragment', undefined, true).then(
+    return this.sendMessageUnreliable_('getFragment', undefined, true).then(
       hash => hash || ''
     );
   }
 
   /**
+   * Update the fragment of the viewer if embedded in a viewer,
+   * otherwise update the page url fragment
+   * @param {string} fragment
+   * @return {!Promise}
+   */
+  updateFragment(fragment) {
+    if (!this.isEmbedded_) {
+      if (this.win.history.replaceState) {
+        const newUrl = this.win.location.href.split('#')[0] + '#' + fragment;
```

Comment:
Do you mean `this.win.history.replaceState({}, '', fragment);` ? That won't work because it will replace the entire url.